### ISSUE TITLE
Add builtin or std.meta fn to detect if field/variable is actually "undefined" at comptime

### ISSUE THREAD

**Issue by frmdstryr** [ziglang/zig#11519]
### Zig Version

0.10.0-dev.1601+109e730c8

### Steps to Reproduce

This is not a bug but after a short discussion on IRQ it was suggested that I create an issue. 

I am attempting to generate an sql table create command based on a struct:

```zig
const std = @import("std");
const comptimePrint = std.fmt.comptimePrint;

const User = struct {
    name: [60:0]u8 = undefined,
};

pub fn createTable(comptime T: type, comptime table_name: []const u8) []const u8 {
    comptime {
         var sql: []const u8 = comptimePrint("CREATE TABLE {s}(\n", .{table_name});
         inline for (std.meta.fields(T))| f | {
            sql = sql ++ comptimePrint("  {s} {}", .{f.name, f.field_type});
            if (f.default_value) |value| {
                const v = @ptrCast(*const f.field_type, value);
                // the name field has a default value but it is "undefined" 
                // so I do not want the database to have a default as
                // the user or database is required to have the value
                // How can it actually detect if 'v' is undefined here?
                //if (!@isUndefined(v)) {
                    sql = sql ++ comptimePrint(" DEFAULT '{s}'", .{v});
                //}
                
            }
            if (@typeInfo(f.field_type) != .Optional) {
                sql = sql ++ " NOT NULL"; // name is not null
            }
            sql = sql ++ ",\n";
         }
         sql = sql ++ ");\n";
         return sql;
    }
}
pub fn main() !void {
    const q = createTable(User, "users");
    std.log.info("{s}", .{q});
}
```

Full code is [here](https://github.com/frmdstryr/zq/blob/master/src/backend/postgres.zig#L940) and model here https://github.com/frmdstryr/zq/blob/master/tests/main.zig#L68

Open to any suggestions on how to better achieve this. 

### Expected Behavior

I would like some way to determine if a value is actually "undefined" at comptime in order to skip setting the default.  Ideally some builtin or something in std.meta should let me detect if a variable is has not been defined.

### Actual Behavior

 If I attempt to check for all 0xAA Zig is doing it's job correctly preventing the comparison and giving a compile error.

**Comment by matu3ba** [comment]
Why does `name: ?[60:0]u8 = null,` not work?

`comptime std.assert(name != null)` should do the trick.

**Comment by frmdstryr** [comment]
Yes optional works fine for columns that are allowed to be nullable. Unfortunately that does not solve how to handle fields/columns that are required to have a value.

**Comment by matu3ba** [comment]
> Unfortunately that does not solve how to handle fields/columns that are required to have a value.

Why do you need a specific value at comptime? 
Please explain, why and how my suggestion does not work with a brief outline of the use case.

**Comment by InKryption** [comment]
@matu3ba The issue (if I understand correctly) is that they have no control over whether the user-supplied field is nullable or not: they want to use the default value of the given field, if available, but have no way to handle if the default value of that field is `undefined` - which, if it is, they either don't want to use that value, or simply want to ignore it.
As it stands, any attempt to use `undefined` in this context results in a compile error, meaning the API is simply unable to handle `undefined` as a default field value - the actual solution would be to make the tradeoff of simply not using the user's default field value at all.
